src/Flow.cpp

```diff
@@ -0,0 +1,19 @@
+#include "Flow.hpp"
+#include "NetworkSimulator.hpp"
+
+#include <stdexcept>
+
+Flow::Flow(const std::vector<std::shared_ptr<NetworkDevice>>& path,
+           const std::vector<std::uint32_t>& distances_ns,
+           std::uint64_t packet_generation_interval_ns):
+        packet_generation_interval_ns(packet_generation_interval_ns) {
+    if (path.size() < 2) {
+        throw std::runtime_error("Error: Incorrect flow path.");
+    }
+    if (distances_ns.size() != path.size() - 1) {
+        throw std::runtime_error("Error: Incorrect flow distances.");
+    }
+    this->path = path;
+    this->distances_ns = distances_ns;
+    id = Flow::last_given_flow_id++;
```

Comment:
Thinking of a flow as a *socket pair*, I suppose the flow id can be hashed from the ids of the source and destination?